lib/API/DX/Device.cpp

```diff
@@ -571,6 +684,12 @@ class DXDevice : public offloadtest::Device {
     return Bundle;
   }
 
+  llvm::Expected<ResourceBundle> createSRV(Resource &R, InvocationState &IS) {
```

Comment:
Another possible approach would be to always use reserved resources, and then just have `tilesMapped == std::nullopt` or whatever result in mapping the entire resource.